### ISSUE TITLE
Legacy Recent Filepaths URI Exception

### ISSUE THREAD

**Issue by ohnocyka** [IsmAvatar/LateralGM#376]
I open up LateralGM 1.8.40

It doesn't even open to anything and comes up with this:'

```
LateralGM Version: 1.8.40

Operating System: Windows 10
Version: 10.0
Architecture: amd64

Java Name: Java HotSpot(TM) 64-Bit Server VM
Java Vendor: Oracle Corporation
Version: 1.8.0_171

Available processors (cores): 4
Free memory (bytes): 68295192
Maximum memory (bytes): 1417150464
Total memory available to JVM (bytes): 96468992

File system root: C:\
Total space (bytes): 499030331392
Free space (bytes): 147759169536
Usable space (bytes): 147759169536

File system root: D:\
Total space (bytes): 104853504
Free space (bytes): 90107904
Usable space (bytes): 90107904

File system root: E:\
Total space (bytes): 1000168484864
Free space (bytes): 996941664256
Usable space (bytes): 996941664256

File system root: G:\
Total space (bytes): 0
Free space (bytes): 0
Usable space (bytes): 0

Stack trace:
java.net.URISyntaxException: Illegal character in opaque part at index 2: C:\Users\Dylan\Desktop\yujju.project.gmx - Shortcut.lnk
	at java.net.URI$Parser.fail(Unknown Source)
	at java.net.URI$Parser.checkChars(Unknown Source)
	at java.net.URI$Parser.parse(Unknown Source)
	at java.net.URI.<init>(Unknown Source)
	at org.lateralgm.components.GmMenuBar.updateRecentFiles(GmMenuBar.java:86)
	at org.lateralgm.components.GmMenuBar.<init>(GmMenuBar.java:139)
	at org.lateralgm.main.LGM.main(LGM.java:1138)

java.lang.IllegalStateException: no splash screen available
	at java.awt.SplashScreen.checkVisible(Unknown Source)
	at java.awt.SplashScreen.update(Unknown Source)
	at org.lateralgm.main.LGM$SplashProgress.update(LGM.java:1440)
	at org.lateralgm.main.LGM$SplashProgress.progress(LGM.java:1433)
	at org.lateralgm.main.LGM.main(LGM.java:1140)
```

**Comment by RobertBColton** [comment]
Ok, I see what's going on, but I am having trouble reproducing it. As you can see from `yujju.project.gmx - Shortcut.lnk` you apparently opened a project through a shortcut. Now it's in your recent files menu.

You can temporarily work around this by clearing your recent files menu and it should go away. I would like to get it fixed, but I am not able to repeat this on my end. The only way i can even select to open a project through a shortcut is by using the "All Files" filter on the open dialog, but even then the recent files menu is rejecting it because no reader can be found.

**Comment by RobertBColton** [comment]
@ohnocyka I will need exact steps to reproduce this on my end before we can fix it. Please provide such steps if you discover any. Thanks.

**Comment by RobertBColton** [comment]
As shown by #442, this apparently occurs without shortcuts as well. Man I would really like to get this one solved soon.

**Comment by RobertBColton** [comment]
From the people experiencing this, if you would like to help in getting it fixed, I need answers to these questions from a few people.

1) What were you doing when the exception occurred?
2) Is this the first time you used LateralGM?
3) Were you using LateralGM by itself or in combination with ENIGMA?
4) Were you or have you ever opened a file by dragging and dropping it onto LGM or passing it from the command line?
5) How do you normally open files in LateralGM?

**Comment by RobertBColton** [comment]
Alright, I figured out what you were people were doing. You were using old LGM versions, like 1.5.7.1 and upgrading. Around that time it was changed in 986260d6538e3fb9ace399bef003d215c9860703 so that the file readers use a URI instead of a file path. So, you people have raw filepath strings in your recent files and new LGM hates them because they aren't URIs.
http://lateralgm.org/download.php

Workaround for all of you right now is simple, follow my steps on #136 to clear the recent files menu from the registry if you can't clear them from LGM itself. I will look at fixing this issue, and we will probably just allow both URIs and regular filepaths.

**Comment by RobertBColton** [comment]
Alright I've sent 385f7f24de904aa8a82664d695de907850481ff3 to fix this and the new LGM 1.8.226 is available for download. Testing seems to confirm the issue is gone and recent files are correctly imported from 1.5.7.1 and usable. Later we may allow both file paths and URIs from the command line and recent files. If you have any other issues, feel free to post new issues! 👍
